### PR TITLE
CI Improvement + import refactoring

### LINKED ISSUE #31: [CI] add diplomacy_research
We want to add the diplomacy research https://github.com/diplomacy/research to the ubuntu instance that is spun up by the CI pipeline (https://github.com/ALLAN-DIP/baseline_bots/blob/main/.github/workflows/CI.yml), so we can automatically test bots like RealPolitik, which depend on this package. One possible solution is creating a Dockerfile and making the tests run within that. See the test_CI branch for a Dockerfile that is close to being finished.

Also see https://github.com/ALLAN-DIP/baseline_bots/issues/29#issuecomment-1261339549_

### PR DESCRIPTION
**Pull request by SiddarGu**
dockerfile and workflow for testing bots on TACC

Resolves #31

### CODE REVIEW

**Comment by trigaten** [comment]
1. Is the CI able to successfully run Dipnet bots?
2. Can you move as many of these files as possible into container/

**Comment by SiddarGu** [comment]
> 

See the run: https://github.com/SiddarGu/baseline_bots/actions/runs/3253118119/jobs/5340071307
It queries TACC and creates a game test_game_114514. If you login and join the game you can see dipnet_TURKEY showing up as turkey.

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,127 @@
...
+COPY run_model_server.sh /model/src/model_server/run_model_server.sh
+RUN chmod 777 /model/src/model_server/run_model_server.sh
+RUN chmod -R 777 /model/src/model_server/bot_neurips2019-sl_model
+
+# TODO: Get this to work for RL model as well
+
+# Clone repos
+RUN git clone https://github.com/SHADE-AI/diplomacy.git
+RUN git clone https://github.com/SHADE-AI/research.git
+RUN git clone --single-branch --branch containerizing https://github.com/ALLAN-DIP/baseline_bots.git
```
Comment (trigaten) [inline Dockerfile:24]: What exactly is this line doing?

Comment (trigaten) [reply Dockerfile:24]: It seems like it is only cloning from this containerizing branch, which is not what we want in a CI pipeline.

File: scripts/dip_ui_bot_launcher.py [hunk 2]
```diff
@@ -6,18 +6,18 @@
...
 from diplomacy.client.connection import connect
 from diplomacy.utils import exceptions
 from diplomacy import Message
 import argparse
-# from bots.dipnet.selectively_transparent_bot import SelectivelyTransparentBot
+# from baseline_bots.bots.dipnet.selectively_transparent_bot import SelectivelyTransparentBot
 
-# from bots.dipnet.transparent_bot import TransparentBot
-# from bots.dipnet.transparent_proposer_bot import TransparentProposerDipBot
+# from baseline_bots.bots.dipnet.transparent_bot import TransparentBot
```
Comment (trigaten) [inline scripts/dip_ui_bot_launcher.py:19]: remove commented out code

File: src/baseline_bots/bots/__init__.py [hunk 3]
```diff
@@ -7,4 +7,5 @@
     RandomHonestOrderAccepterBot,
 )
 from baseline_bots.bots.random_proposer_bot import RandomProposerBot
-from baseline_bots.bots.smart_order_accepter_bot import SmartOrderAccepterBot
+
+# from baseline_bots.bots.smart_order_accepter_bot import SmartOrderAccepterBot
```
Comment (trigaten) [inline src/baseline_bots/bots/__init__.py:11]: remove commented out code

File: src/baseline_bots/utils.py [hunk 4]
```diff
@@ -97,44 +106,107 @@ def FCT(string) -> str:
...
 
 def parse_PRP(msg) -> str:
     """Detaches PRP from main arrangement"""
     if "PRP" not in msg:
         raise ParseError("This is not an PRP message")
     try:
-        return msg[5:-1]
+        return msg[msg.find("(") + 1:-1]
     except Exception:
-        raise ParseError(f"Cant parse PRP msg {msg}")
```
Comment (trigaten) [inline src/baseline_bots/utils.py:117]: Why make this change?

Comment (SiddarGu) [reply src/baseline_bots/utils.py:117]: I think this is from merging ALLAN-DIP/main. The only thing I changed in baseline_bots is removing import REJ

**Pull request merged**